model/GuiSettingsService.php

```diff
@@ -49,8 +49,31 @@ class GuiSettingsService extends ConfigurableService
      */
     const PROCTORING_ALLOW_PAUSE = 'canPause';
 
-    public function __construct(array $options = array())
+    const OPTION_SHOW_COLUMN_FIRST_NAME = 'showColumnFirstName';
+    const OPTION_SHOW_COLUMN_LAST_NAME = 'showColumnLastName';
+    const OPTION_SHOW_COLUMN_AUTHORIZE = 'showColumnAuthorize';
+    const OPTION_SHOW_COLUMN_REMAINING_TIME = 'showColumnRemainingTime';
+    const OPTION_SHOW_COLUMN_EXTENDED_TIME = 'showColumnExtendedTime';
+    const OPTION_SHOW_COLUMN_CONNECTIVITY = 'onlineStatus';
+
+    const OPTION_SHOW_ACTION_SHOW_HISTORY = 'showActionShowHistory';
+
+    /**
+     * @return array
+     */
+    public function asArray()
     {
-        parent::__construct($options);
+        return [
+            self::PROCTORING_REFRESH_BUTTON         => $this->hasOption(self::PROCTORING_REFRESH_BUTTON) ? $this->getOption(self::PROCTORING_REFRESH_BUTTON) : true,
+            self::PROCTORING_AUTO_REFRESH           => $this->hasOption(self::PROCTORING_AUTO_REFRESH) ? $this->getOption(self::PROCTORING_AUTO_REFRESH) : true,
+            self::PROCTORING_ALLOW_PAUSE            => $this->hasOption(self::PROCTORING_ALLOW_PAUSE) ? $this->getOption(self::PROCTORING_ALLOW_PAUSE) : true,
+            self::OPTION_SHOW_COLUMN_FIRST_NAME     => $this->hasOption(self::OPTION_SHOW_COLUMN_FIRST_NAME) ? $this->getOption(self::OPTION_SHOW_COLUMN_FIRST_NAME) : true,
+            self::OPTION_SHOW_COLUMN_LAST_NAME      => $this->hasOption(self::OPTION_SHOW_COLUMN_LAST_NAME) ? $this->getOption(self::OPTION_SHOW_COLUMN_LAST_NAME) : true,
+            self::OPTION_SHOW_COLUMN_AUTHORIZE      => $this->hasOption(self::OPTION_SHOW_COLUMN_AUTHORIZE) ? $this->getOption(self::OPTION_SHOW_COLUMN_AUTHORIZE) : true,
+            self::OPTION_SHOW_COLUMN_REMAINING_TIME => $this->hasOption(self::OPTION_SHOW_COLUMN_REMAINING_TIME) ? $this->getOption(self::OPTION_SHOW_COLUMN_REMAINING_TIME) : true,
+            self::OPTION_SHOW_COLUMN_EXTENDED_TIME  => $this->hasOption(self::OPTION_SHOW_COLUMN_EXTENDED_TIME) ? $this->getOption(self::OPTION_SHOW_COLUMN_EXTENDED_TIME) : true,
+            self::OPTION_SHOW_COLUMN_CONNECTIVITY   => $this->hasOption(self::OPTION_SHOW_COLUMN_CONNECTIVITY) ? $this->getOption(self::OPTION_SHOW_COLUMN_CONNECTIVITY) : true,
```

Comment:
I think 'true' is not the right default value,  because of
https://github.com/oat-sa/extension-tao-proctoring/blob/e4ee3e2b9c2f327d14ad7c50719a338900a375bf/views/js/controller/Delivery/monitoring.js#L619

Reply:
This is right default value.
Both BE and FE are using boolean value in that case.

Reply:
it is not about boolean, it is about logic
On frontend, if we don't have onlineStatus then it will be false (allowedConnectivity = data.onlineStatus || false)
On backend,  if we don't have onlineStatus then it will be true (self::OPTION_SHOW_COLUMN_CONNECTIVITY   => $this->hasOption(self::OPTION_SHOW_COLUMN_CONNECTIVITY) ? $this->getOption(self::OPTION_SHOW_COLUMN_CONNECTIVITY) : true)

Reply:
fixed